pinot-spi/src/main/java/org/apache/pinot/spi/config/table/IndexingConfig.java

```diff
@@ -53,6 +53,8 @@
   private SegmentPartitionConfig _segmentPartitionConfig;
   private boolean _aggregateMetrics;
   private boolean _nullHandlingEnabled;
+  private boolean _optimizeDictionaryEnabled;
+  private double _thresholdMinPercentDictionaryStorageSaved;
```

Comment:
Suggest changing it to `_dictionarySizeRatioThreshold` (feel free to rename), and make it default to 1.0. In certain cases we might prefer raw index even if its size is slightly larger because that can avoid the two hops lookup, and provide better locality.

Reply:
Just confirming - we are not changing the default behavior (dictionary on all columns) yet right ?

Reply:
No, we are not changing that. @Jackie-Jiang the default value doesn't make sense to me based on the current formula.  We are creating dictionaries by default and the logic only avoid creating it when we can save some significant storage.
If a column is set as rawValueColumn then no dictionary will be created, irrespective of the logic here.

Reply:
@KKcorps Understood. What I meant is that let's keep the absolute ratio instead of the saved ratio because in certain cases we might want to generate raw index even if the size is slightly larger, and configuring negative saved ratio can be confusing. To keep the same behavior, we may make the default 0.85

Reply:
cool. Changed the logic to ratios now.